core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/IdentityProvider.java

```diff
@@ -22,105 +22,106 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.b2international.commons.exceptions.BadRequestException;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
 import com.b2international.snowowl.core.events.util.Promise;
+import com.b2international.snowowl.core.setup.Environment;
 
 /**
  * @since 5.11
  */
 public interface IdentityProvider {
 
 	/**
-	 * Global logger that should be used for identity provider related logging. 
+	 * Global logger that should be used for identity provider related logging.
 	 */
 	Logger LOG = LoggerFactory.getLogger("identity");
-	
+
 	/**
-	 * @since 7.0 
+	 * @since 7.0
 	 */
-	IdentityProvider NOOP = new IdentityProvider() {
-		
+	IdentityProvider UNPROTECTED = new IdentityProvider() {
+
 		@Override
 		public User auth(String username, String password) {
 			return new User(username, List.of(Permission.ADMIN));
 		}
-		
+
+		@Override
+		public User authJWT(String token) {
+			final DecodedJWT decodedToken = JWT.decode(token);
```

Comment:
Please add a comment here that tells readers that this identity provider has no means of verifying JWT contents.